src/index.js

```diff
@@ -1447,5 +1452,266 @@ module.exports = {
         return expect.shift(parseXml(subject));
       }
     );
+
+    function scoreElementAgainstSpec(element, spec) {
+      const isTextMatching = (value, valueSpec) => {
+        if (typeof valueSpec === 'string') {
+          if (value === valueSpec) {
+            return true;
+          }
+        } else if (valueSpec instanceof RegExp) {
+          if (valueSpec.test(value)) {
+            return true;
+          }
+        } else if (typeof valueSpec === 'function') {
+          return true;
+        }
+
+        return false;
+      };
+      const isHtml = isInsideHtmlDocument(element);
+
+      let score = 0;
+
+      const nodeName = isHtml
+        ? element.nodeName.toLowerCase()
+        : element.nodeName;
+
+      if (isTextMatching(nodeName, spec.name)) {
```

Comment:
This will come out as false in HTML mode if `spec.name` is in uppercase?

Reply:
I think I'll always compare case nonsensitive as this is just given points for similarity. I'll change that.